tests/acceptance/demo-page/dark-mode-toggle-test.js

```diff
@@ -0,0 +1,82 @@
+import { module, test } from 'qunit';
+import { currentURL } from '@ember/test-helpers';
+import { setupApplicationTest } from 'codecrafters-frontend/tests/helpers';
+import demoPage from 'codecrafters-frontend/tests/pages/demo-page';
+import { timeout } from 'ember-concurrency';
+
+module('Acceptance | demo page | dark-mode-toggle', function (hooks) {
+  setupApplicationTest(hooks);
+
+  test('it lands on a correct URL', async function (assert) {
+    await demoPage.demoTabs.darkModeToggle.visit();
+    assert.strictEqual(currentURL(), '/demo/dark-mode-toggle', 'URL is correct');
```

Comment:
Same as above, would try to simplify and keep just one short test

Reply:
Same here, collapsed most of tests into one, compacted all the tests and left only 4 most essential ones (logically grouped by what they're testing)